ls1intum/Artemis
# [Bugfix] Only fetch categories if static code analysis is enabled

### Checklist

### Motivation and Context


The grading page should only request the categories if static code analysis is enabled.

### Steps for Testing


1. Log in to Artemis
2. Create a programming exercise with and without static code analysis
3. Open your browser developer tools
4. Navigate to Grading
--> Exercise without SCA: No request to fetch the categories should be executed
--> Exercise with SCA: Request to fetch the categories should be executed. Categories should be displayed

### Test Coverage

Nothing changed in terms of coverage
